lib/grape/validations/types/custom_type_coercer.rb

```diff
@@ -136,9 +136,13 @@ def infer_type_check(type)
             # Note that this will fail unless a method is also
             # passed, or if the type also implements a parse() method.
             type
+          elsif type == [Integer]
```

Comment:
This seems like a 1-off, shouldn't it be that `type.is_a?(Enumerable)` of 1 simple type, and then it's `value.all?` of that type?